src/server/scripts/EasternKingdoms/Stratholme/boss_baron_rivendare.cpp

```diff
@@ -16,170 +15,157 @@
  * with this program. If not, see <http://www.gnu.org/licenses/>.
  */
 
-/* ScriptData
-SDName: Boss_Baron_Rivendare
-SD%Complete: 70
-SDComment: aura applied/defined in database
-SDCategory: Stratholme
-EndScriptData */
-
 #include "ScriptMgr.h"
 #include "InstanceScript.h"
 #include "ScriptedCreature.h"
+#include "SpellInfo.h"
 #include "stratholme.h"
 
-enum Says
+enum Texts
 {
-    SAY_BARON_RUN_START         = 0,
-    SAY_BARON_RUN_BOSS_KILL     = 1,
-    SAY_BARON_RUN_FAIL          = 2,
-    SAY_EVENT_RAMSTEIN          = 3,
-    SAY_EVENT_BARON             = 4
+    EMOTE_RAISE_DEAD    = 0,    // %s raises an undead servant back to life!
+    EMOTE_DEATH_PACT    = 1     // %s attempts to cast Death Pact on his servants!     
 };
 
 enum Spells
 {
-    SPELL_SHADOWBOLT            = 17393,
-    SPELL_CLEAVE                = 15284,
-    SPELL_MORTALSTRIKE          = 15708,
-
-    SPELL_UNHOLY_AURA           = 15284,
-    SPELL_RAISEDEAD             = 15708,  //triggers death pact (17471)
-
-    SPELL_RAISE_DEAD1           = 17475,
-    SPELL_RAISE_DEAD2           = 17476,
-    SPELL_RAISE_DEAD3           = 17477,
-    SPELL_RAISE_DEAD4           = 17478,
-    SPELL_RAISE_DEAD5           = 17479,
-    SPELL_RAISE_DEAD6           = 17480,
+    SPELL_SHADOWBOLT    = 17393,
+    SPELL_CLEAVE        = 15284,
+    SPELL_MORTALSTRIKE  = 15708,
+    SPELL_DEATH_PACT_1  = 17698, // Heal Rivendare and damage skeleton
+    SPELL_DEATH_PACT_2  = 17471, // Visual effect
+    SPELL_DEATH_PACT_3  = 17472, // Instant kill self only
+    SPELL_RAISE_DEAD    = 17473, // Inits. 17698 and 17471
+    SPELL_UNHOLY_AURA   = 17467
 };
 
-// Define Add positions
-Position const ADD_POS_1 = {4017.403809f, -3339.703369f, 115.057655f, 5.487860f};
-Position const ADD_POS_2 = {4013.189209f, -3351.808350f, 115.052254f, 0.134280f};
-Position const ADD_POS_3 = {4017.738037f, -3363.478016f, 115.057274f, 0.723313f};
-Position const ADD_POS_4 = {4048.877197f, -3363.223633f, 115.054253f, 3.627735f};
-Position const ADD_POS_5 = {4051.777588f, -3350.893311f, 115.055351f, 3.066176f};
-Position const ADD_POS_6 = {4048.375977f, -3339.966309f, 115.055222f, 2.457497f};
+enum BaronRivendareEvents
+{
+    EVENT_SHADOWBOLT    = 1,
+    EVENT_CLEAVE        = 2,
+    EVENT_MORTALSTRIKE  = 3,
+    EVENT_RAISE_DEAD    = 4
+};
 
-class boss_baron_rivendare : public CreatureScript
+// Raise Dead
+uint32 const RaiseDeadSpells[6] =
+{
+    17475, 17476, 17477,
+    17478, 17479, 17480
+};
+
+struct boss_baron_rivendare : public BossAI
 {
 public:
-    boss_baron_rivendare() : CreatureScript("boss_baron_rivendare") { }
+    boss_baron_rivendare(Creature* creature) : BossAI(creature, TYPE_BARON), RaiseDead(false) { }
 
-    CreatureAI* GetAI(Creature* creature) const override
+    void Reset() override
     {
-        return GetStratholmeAI<boss_baron_rivendareAI>(creature);
+        // DoCastSelf doesn't add aura after reset
+        me->AddAura(SPELL_UNHOLY_AURA, me);
```

Comment:
Why?
17467 Unholy Aura has a ``` SPELL_AURA_PERIODIC_TRIGGER_SPELL ``` that trigger each 2 secs and is a self cast.
Is better try debug to find why it's not working (if you cant debug, i can help tomorrow)

Reply:
Yeah sure would be great.